### DISCUSSION TITLE
How to create a random shaped border around image?

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#5400] [Digital Image Processing]
Version: ImageMagick 7.1.0-35 Q16-HDRI x86_64 20062 https://imagemagick.org
Features: Cipher DPC HDRI OpenMP(4.5) 
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lzma pangocairo png tiff webp x xml zlib
Compiler: gcc (8.5)

I have many cliparts, for which I want to create a random shaped borders (smooth contours). 

![randomborder-1](https://user-images.githubusercontent.com/12046302/182060380-762adbb6-f88e-432d-8c83-e8274d24a2d0.png)
![randomborder-2](https://user-images.githubusercontent.com/12046302/182060384-5d0110a8-7d71-430e-abc0-3500a4f61eb7.png)

Any help is appreciated.

Thanks,

**Comment by snibgo** [comment]
Those shapes don't seem to be random. If you think they are, please say what about them is random.

**Comment by srini8080** [comment]
Any randomness around the picture is fine. But smooth contours. 

Thanks

**Comment by srini8080** [comment]
Something like BLOBs what Fred created in his scripts. Then given Image inside it. Entire combination cut to the border of the BLOB to make a PNG with transparent background

![image](https://user-images.githubusercontent.com/12046302/182082580-28407523-9646-4ca9-bce6-19db9708b943.png)

**Reply by fmw42** [reply]
Can you not just use my randomblob script? Invert the result and use it to mask your image.

**Reply by srini8080** [reply]
Let me try.

Thanks

**Comment by srini8080** [comment]
#!/bin/sh
set -x
set -e

# usage $0 imagefile contourfile/maskfile

imagefile="$1"
masksize=`convert ${2} -format "%wx%h" info:`
convert \( "$imagefile" -resize $masksize -background white -gravity center -extent $masksize \) ${2} -compose copy_opacity -composite -compose over -background transparent -flatten png:result.png

convert "$2" "$1" \
   +distort SRT "0,0 %[fx:t?max(u.w/v.w,u.h/v.h):1] 0 0,0" \
   -shave 1x1 +repage -gravity center -compose srcin -composite \
   -compose over -background white -flatten result.jpg

used on the files is cutting the image. I want the image fully inside the mask.

Thanks

![result](https://user-images.githubusercontent.com/12046302/182619796-7fbe7c96-c812-48bf-b8d3-e0b21a9b50cb.jpg)
![result](https://user-images.githubusercontent.com/12046302/182619798-29caa0a5-1525-4630-aab7-508699ef4b75.png)
![contour-11](https://user-images.githubusercontent.com/12046302/182619804-3227a2dd-436b-4737-a97d-1ca18588d989.png)
![35666](https://user-images.githubusercontent.com/12046302/182619806-d772c0e1-b02b-4502-a251-d6d3d3c4cd12.jpg)

**Comment by srini8080** [comment]
getting the height and the width of irregular mask?

**Reply by fmw42** [reply]
Please ask your question clearer. I do not understand what you want. The mask is resulting in the image only inside the mask when properly composited.  What do you mean by height and width of mask?  You can simply -trim it and get the dimensions of the trimmed blob in the mask.

**Comment by srini8080** [comment]
Yes, that will help me. Trim and get the blob image dimensions. I was thinking complex math :)

⁣Sent from BlueMail ​

On Aug 5, 2022, 8:32 AM, at 8:32 AM, Fred Weinhaus ***@***.***> wrote:
>Please ask your question clearer. I do not understand what you want.
>The mask is resulting in the image only inside the mask when properly
>composited.  What do you mean by height and width of mask?  You can
>simply -trim it and get the dimensions of the trimmed blob in the mask.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/ImageMagick/ImageMagick/discussions/5400#discussioncomment-3330019
>You are receiving this because you authored the thread.
>
>Message ID:
***@***.***>
